scripts/content-utils/card-sequence.ts

```diff
@@ -0,0 +1,142 @@
+import { GameWorldModifier } from "../../swipeforfuture.com/src/game/ContentTypes"
+import {
+    WorldQuery,
+    CardData,
+    cardRef,
+    setModifier,
+    CardTree,
+    cardsFromTree,
+    isCardTree,
+    combineWorldQueries,
+} from "./"
+
+/**
+ * TODO:
+ *
+ * 1) ✅ Create CardSequence with support for linking cards
+ * - ✅ fix bug to only show final card in sequence once
+ * - (maybe) fix removing unused flags after they have been used to trigger cards in the sequence
+ *
+ * 2) ✅ Add support to execute different modifiers on left and right (implemented by CardTree)
+ *
+ * 3) ✅ Add support for CardTree within the CardSequenuce. If a node is a CardTree, then recursively call the cardsFromTree() function to build that subtree.
+ * 4) Add support for CardSequence within the CardTree. If a node is a CardSequence, then then recursively call the cardsFromSequence() function to build that subsequence.
+ *
+ * 5) Add support for combining multiple CardSequences within a CardSequence
+ */
+
+export interface CardSequence {
+    sequence: (CardTree | CardSequence)[]
+}
+
+export function isCardSequence(node: any): node is CardSequence {
```

Comment:
This type check is not implemented properly. If you take `any` as argument you need to check a lot more things. You should use `isCardSequence(node: CardSequence | CardTree)` instead.

Reply:
```
export function isCardSequence(node: CardTree | CardSequence): node is CardSequence {
    return "sequence" in node
}
```